code/handlers/ldap.q

```diff
@@ -84,7 +83,7 @@ login:{[user;pass]                                              / validate login
     .[.ldap.bind;(.ldap.sessionID;dict);enlist[`ReturnCode]!enlist -2i]                 / attempt authentication
   ];
  
-  `.ldap.cache upsert (user;np;`$.ldap.server;.ldap.port;.z.p; $[0=authorised[`ReturnCode];0;1+0^incache`attempts] ;authorised[`ReturnCode]~0i;0b);  / upsert details of current attempt
+  `.ldap.cache upsert (user;np;`$.ldap.server;.ldap.port;.z.p; 1+0^incache`attempts ;authorised[`ReturnCode]~0i;0b);  / upsert details of current attempt
```

Comment:
Is this change correct? Do we want to keep track of successful attempts? I think this was probably intended to track failed attempts

Reply:
I had changed it because the counter for attempts would always be zero for authenticated users, whereas the change now just counts the number of attempts as it did for the failed attempts, although if only counting failed attempts is the idea then the change isn't necessary

Reply:
I think it's probably just for counting failed attempts, so a user can be locked out after X failed attempts. WIth it like this, someone could successfully login 100 times, fail once and be blocked, which I don't think we want

Reply:
That makes sense actually, counting failed attempts only then